src/callcc/jumper.ts

```diff
@@ -194,6 +201,211 @@ function tryCatchCaptureLogic(path: NodePath<t.Expression | t.Statement>, restor
+    [t.thisExpression(), ...(<any>funParams)]);
+  const stackFrame = t.objectExpression([
+    t.objectProperty(t.identifier('kind'), t.stringLiteral('rest')),
+    t.objectProperty(
+      t.identifier('f'),
+      t.arrowFunctionExpression([], reapply)),
+    t.objectProperty(t.identifier('locals'),
+      t.arrayExpression(<any>locals)),
+    t.objectProperty(t.identifier('index'), applyLbl),
+  ]);
+
+  const ifStmt = t.ifStatement(
+    isNormalMode,
+    t.blockStatement([
+      t.expressionStatement(t.callExpression(pushEagerStack, [
+        stackFrame,
+      ])),
+      nodeStmt,
+      t.expressionStatement(t.callExpression(shiftEagerStack, [])),
+    ]),
+    t.ifStatement(
+      t.binaryExpression('===', target, applyLbl),
+      t.blockStatement([
+        restoreCall(),
+        t.expressionStatement(t.callExpression(shiftEagerStack, [])),
+      ])));
+  (<any>ifStmt).isTransformed = true;
+
+  const ifApply = t.callExpression(t.arrowFunctionExpression([],
+    t.blockStatement([ifStmt])), []);
+  const stmtParent = path.getStatementParent();
+  path.isStatement() ?
+  (path.replaceWith(ifStmt), path.skip()) :
+  t.isStatement(path.parent) ?
+  (stmtParent.replaceWith(ifStmt), stmtParent.skip()) :
+  (path.replaceWith(ifApply), path.skip());
+}
+
+function retvalCaptureLogic(path: NodePath<t.Expression | t.Statement>, restoreCall: () => t.Statement): void {
+  const applyLbl = t.numericLiteral(getLabels(path.node)[0]);
+  const ret = path.scope.generateUidIdentifier('ret');
+
+  const funParent = <NodePath<FunctionT>>path.findParent(p =>
+    p.isFunctionExpression() || p.isFunctionDeclaration());
+  const funId = funParent.node.id, funParams = funParent.node.params,
+  funBody = funParent.node.body;
+
+  const afterDecls = funBody.body.findIndex(e =>
+    !(<any>e).__boxVarsInit__ && !(<any>e).lifted);
+  const { pre, post } = split(funBody.body, afterDecls);
+
+  const locals: t.LVal[] = [];
+  pre.forEach(decls => {
+    if (t.isVariableDeclaration(decls)) {
+      decls.declarations.forEach(x => locals.push(x.id))
+    }
+  });
+  for (const x of Object.keys(funParent.scope.bindings)) {
+    // Type definition is missing this case.
+    if (<string>(path.scope.getBinding(x).kind) !== 'hoisted') {
+      continue;
+    }
+    locals.push(path.scope.getBinding(x).identifier);
+  }
+
+  const reapply = t.callExpression(t.memberExpression(funId, t.identifier("call")),
+    [t.thisExpression(), ...(<any>funParams)]);
+  const stackFrame = t.objectExpression([
+    t.objectProperty(t.identifier('kind'), t.stringLiteral('rest')),
+    t.objectProperty(
+      t.identifier('f'),
+      t.arrowFunctionExpression([], reapply)),
+    t.objectProperty(t.identifier('locals'),
+      t.arrayExpression(<any>locals)),
+    t.objectProperty(t.identifier('index'), applyLbl),
+  ]);
+
+  const nodeBlock: t.Statement[] = t.isReturnStatement(path.node) || t.isThrowStatement(path.node) ?
```

Comment:
When creating AST in this way, it helps if you add a comment saying what the generated code looks like